datafusion/common/src/scalar.rs

```diff
@@ -3885,6 +3889,77 @@ mod tests {
         );
     }
 
+    #[test]
+    fn scalar_try_from_array_list_array_null() {
+        let list = ListArray::from_iter_primitive::<Int32Type, _, _>(vec![
+            Some(vec![Some(1), Some(2)]),
+            None,
+        ]);
+
+        let non_null_list_scalar = ScalarValue::try_from_array(&list, 0).unwrap();
```

Comment:
Previously, for a ScalarValue NULL value from ListArray, its datatype is `DataType::Null`. That's is not correct.